cmd/flux/build_artifact.go

```diff
@@ -66,18 +72,47 @@ func buildArtifactCmdRun(cmd *cobra.Command, args []string) error {
 		return fmt.Errorf("invalid path %q", buildArtifactArgs.path)
 	}
 
-	if _, err := os.Stat(buildArtifactArgs.path); err != nil {
-		return fmt.Errorf("invalid path '%s', must point to an existing directory or file", buildArtifactArgs.path)
+	path := buildArtifactArgs.path
+	var err error
+	if buildArtifactArgs.path == "-" {
+		path, err = saveStdinToFile()
+		if err != nil {
+			return err
+		}
+
+		defer os.Remove(path)
+	}
+
+	if _, err := os.Stat(path); err != nil {
+		return fmt.Errorf("invalid path '%s', must point to an existing directory or file", path)
 	}
 
-	logger.Actionf("building artifact from %s", buildArtifactArgs.path)
+	logger.Actionf("building artifact from %s", path)
 
 	ociClient := oci.NewLocalClient()
-	if err := ociClient.Build(buildArtifactArgs.output, buildArtifactArgs.path, buildArtifactArgs.ignorePaths); err != nil {
+	if err := ociClient.Build(buildArtifactArgs.output, path, buildArtifactArgs.ignorePaths); err != nil {
 		return fmt.Errorf("bulding artifact failed, error: %w", err)
 	}
 
 	logger.Successf("artifact created at %s", buildArtifactArgs.output)
-
 	return nil
 }
+
+func saveStdinToFile() (string, error) {
+	b, err := io.ReadAll(bufio.NewReader(os.Stdin))
+	if err != nil {
+		return "", err
+	}
+	b = bytes.TrimRight(b, "\r\n")
+	f, err := os.CreateTemp("", "*.yaml")
```

Comment:
How is this file placed in the tarball? Does it contain only the file or it has all the tmp dir path also?

Reply:
It has all the `tmp` dir path too.

I considered updating the `Build` in `fluxcd/pkg` so that it puts only the file if an absolute path is passed in but that conflicts with the behaviour when a relative path is passed (it has the full dir path too)

Reply:
If a single file is passed (either via stdin or file path) the tarball should contain only that file in the root. Please make the necessary changes in pkg. Thanks!